### ISSUE TITLE
IndexFollowingIT#testFollowIndexWithConcurrentMappingChanges fails

### ISSUE THREAD

**Issue by danielmitterdorfer** [elastic/elasticsearch#59273]
**Build scan**: https://gradle-enterprise.elastic.co/s/7zvpy3v4eu3ok

**Repro line**:

```
./gradlew ':x-pack:plugin:ccr:internalClusterTest' --tests "org.elasticsearch.xpack.ccr.IndexFollowingIT.testFollowIndexWithConcurrentMappingChanges" \
  -Dtests.seed=A15C3DF4E9421E31 \
  -Dtests.security.manager=true \
  -Dtests.locale=uk-UA \
  -Dtests.timezone=Africa/Ndjamena \
  -Druntime.java=11
```

**Reproduces locally?**: no

**Applicable branches**: 

* `master`
* `7.x`

**Failure history**: Failed [8 times within the last 6 months](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-6M,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'project:elasticsearch%20AND%20test:%22testFollowIndexWithConcurrentMappingChanges%22%20AND%20class:%22org.elasticsearch.xpack.ccr.IndexFollowingIT%22%20AND%20stacktrace:%22java.lang.AssertionError:%20timed%20out%20waiting%20for%20green%20state%22'),sort:!(time,desc))) (two times out of this on a PR build)

**Failure excerpt**:

```
java.lang.AssertionError: timed out waiting for green state
	at __randomizedtesting.SeedInfo.seed([A15C3DF4E9421E31:862EA90C40E1F715]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.elasticsearch.xpack.CcrIntegTestCase.ensureColor(CcrIntegTestCase.java:338)
	at org.elasticsearch.xpack.CcrIntegTestCase.ensureFollowerGreen(CcrIntegTestCase.java:312)
	at org.elasticsearch.xpack.CcrIntegTestCase.ensureFollowerGreen(CcrIntegTestCase.java:307)
	at org.elasticsearch.xpack.ccr.IndexFollowingIT.testFollowIndexWithConcurrentMappingChanges(IndexFollowingIT.java:289)
```

**Comment by dimitris-athanasiou** [comment]
We got another failure on this: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=fedora-29&&immutable/129/console

Unfortunately the build scan links don't seem to work.

**Comment by dnhatn** [comment]
Fixed in #60534.
